include/chemfiles/files/BinaryFile.hpp

```diff
@@ -161,6 +161,8 @@ class BinaryFile: public File {
         this->read_f32(&value, 1);
         return value;
     }
+    /// Read exactly as many 32-bit floating point numbers as fit in the pre-allocated vector
+    void read_f32_array(std::vector<float>& data) { this->read_f32(data.data(), data.size()); }
```

Comment:
I think we should provide these for all supported types, not only f32/f64. About the function name, It feels like `read_f32_array` and `read_f32` are similar enough to have the same name & use an overloaded function to distinguish between them. What do you think?

Reply:
Agree, it makes sense to have overloaded functions and it cuts down on typing. I'm not to sure how to deal with overloading pure virtual functions though.